common/ui/circuit-overlay/src/main/kotlin/app/tivi/overlays/BottomSheetOverlay.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.tivi.overlays
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material3.BottomSheetDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.isSpecified
+import androidx.compose.ui.unit.Dp
+import com.slack.circuit.foundation.CircuitContent
+import com.slack.circuit.overlay.Overlay
+import com.slack.circuit.overlay.OverlayHost
+import com.slack.circuit.overlay.OverlayNavigator
+import com.slack.circuit.runtime.Screen
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+class BottomSheetOverlay<Model : Any, Result : Any>(
```

Comment:
Do you think we should offer a batteries-included component like this? Definitely know there's like 4 different implementations of this same idea floating around different repos at the moment heh

Reply:
Yeah I was thinking the same thing. Probably worth keeping as a separate artifact. It also gets tricky with Jetpack Compose vs Compose Multiplatform.